projects/plugins/jetpack/extensions/blocks/story/story.php

```diff
@@ -88,6 +89,31 @@ function ( $media_file ) {
 						'caption' => wp_get_attachment_caption( $attachment_id ),
 					)
 				);
+
+				// Set the poster attribute for the video tag if a poster image is available.
+				if ( ! empty( $video_meta['videopress']['poster'] ) ) {
+					$poster_url = $video_meta['videopress']['poster'];
+				} elseif ( ! empty( $video_meta['thumb'] ) ) {
+					$video_url  = wp_get_attachment_url( $attachment_id );
+					$poster_url = str_replace( wp_basename( $video_url ), $video_meta['thumb'], $video_url );
+				}
+
+				if ( $poster_url ) {
+					$poster_width  = esc_attr( $media_file['width'] );
+					$poster_height = esc_attr( $media_file['height'] );
+					$content_width = Jetpack::get_content_width();
+					if ( is_numeric( $content_width ) ) {
+						$poster_height = round( ( $content_width * $poster_height ) / $poster_width );
+						$poster_width  = $content_width;
+					}
+					$media_file = array_merge(
+						$media_file,
+						array(
+							'poster' => $poster_url . '?resize=' . rawurlencode( $poster_width . ',' . $poster_height ),
```

Comment:
I feel like there should be a better approach than using `rawurlencode` here but I couldn't find one, feedback very welcome 🙏 .

The reason I added it (here and in the static fallback below) is because without it sometimes two requests would fire (one from the static fallback, and one from the `poster` attribute of the `video` element once JS loaded), and it didn't look like the browser necessarily recognized them as the same request:

https://videos.files.wordpress.com/kFq0ZBzi/wp_story1616037408131_0_mp4_hd.original.jpg?resize=580,1031
https://videos.files.wordpress.com/kFq0ZBzi/wp_story1616037408131_0_mp4_hd.original.jpg?resize=580%2C1031

The behavior was different on WP.com and Jetpack, I believe because they have separate post content filtering/image resizing logic. Encoding that portion of the poster URL everywhere was the only way I found to prevent double loading on both Jetpack and WP.com.

Reply:
I think `rawurlencode` should work fine here. Would it be worth building the URL with `add_query_arg`?

Reply:
I think that makes sense, done in 5d935be 👍